### PR TITLE
Review book

### PR DESCRIPTION
**Pull request by ngoctt-0375**
## Related Tickets
https://edu-redmine.sun-asterisk.vn/issues/67401
## WHAT (optional)
- Thêm sửa xóa bình luận về sách

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)

[Screencast from 11-09-2023 09:58:17.webm](https://github.com/awesome-academy/ruby_naitei19_library_management/assets/142281496/6e0144f5-863a-4016-9015-79d76b80f188)

### CODE REVIEW

**Comment by ngoctt-0375** [comment]
ready

**Comment by ngoctt-0375** [comment]
ready

File: app/controllers/reviews_controller.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  before_action :logged_in_user
+  before_action :load_book
+  before_action :load_review
+  def new
+    @review = Review.new
+  end
+
+  def create
+    @review = Review.new(review_params)
+    @review.book_id = @book.id
```
Comment (vanvtt-0952) [inline app/controllers/reviews_controller.rb:11]:
```suggestion
    @review = @book.reviews.new review_params
```

File: app/controllers/reviews_controller.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
+class ReviewsController < ApplicationController
+  before_action :logged_in_user
+  before_action :load_book
+  before_action :load_review
+  def new
+    @review = Review.new
```
Comment (vanvtt-0952) [inline app/controllers/reviews_controller.rb:6]:
```suggestion
    @review = current_user.reviews.new
```

File: app/controllers/reviews_controller.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+  def load_review
+    @review = current_user.reviews.find_by book_id: @book.id
+  end
+
+  def update_average_rating
+    @book.reload
+    reviews = @book.reviews
+    total_rating = reviews.sum(:rating)
+    @book.update(average_rating: total_rating.to_f / reviews.count)
```
Comment (vanvtt-0952) [inline app/controllers/reviews_controller.rb:64]: ri thì tìm hieru về transaction đi bạn nếu update rate ko thành công thì rollback data cho review chứ

**Comment by ngoctt-0375** [comment]
ready

**Pull request merged**